addons/sale/models/sale.py

```diff
@@ -1620,6 +1620,61 @@ def product_id_change(self):
 
         return result
 
+    def _get_computed_price_unit(self, product=None):
+        ''' Helper to get the default price unit based on the product by taking care of the taxes
+        set on the product and the fiscal position.
+        :return: The price unit.
+        '''
+        self.ensure_one()
+
+        if not self.product_id:
+            return 0.0
+
+        company = self.order_id.company_id
+        currency = self.order_id.currency_id
+        company_currency = company.currency_id
+        fiscal_position = self.order_id.fiscal_position_id
+        is_refund_document = self.product_uom_qty < 0
```

Comment:
this is not necessary. This is used for taxes management but, in your case, you are still a sale document even with a negative quantity